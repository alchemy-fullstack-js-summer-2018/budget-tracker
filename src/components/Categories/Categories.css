:local(.categories) {

  h1 {
    color: #2c3e50;
  }

  .category-form {
    color: green;
  }

}