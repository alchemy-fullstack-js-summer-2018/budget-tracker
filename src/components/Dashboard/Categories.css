p {
  color: seagreen;
  font-weight: bolder;
}