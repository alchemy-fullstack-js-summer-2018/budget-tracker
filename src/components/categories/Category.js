import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import CategoryItem from './CategoryItem';
import CategoryForm from './CategoryForm';
import { update, remove } from './actions';
import styles from './Category.css';

class Category extends PureComponent {

  state = {
    editing: false
  };

  static propTypes = {
    category: PropTypes.object.isRequired,
    update: PropTypes.func.isRequired,
    remove: PropTypes.func.isRequired
  };

  toggleEdit = () => {
    console.log('toggling');
    this.setState(({ editing }) => ({ editing: !editing }));
  };

  handleComplete = category => {
    this.props.update(category);
    this.toggleEdit();
  };

  handleRemove = key => {
    this.props.remove(key);
    this.toggleEdit();
  };

  render() { 
    const { category } = this.props;
    const { editing } = this.state;

    return (
      <li className={styles.category}>
        {editing
          ? <CategoryForm
            category={category}
            onComplete={this.handleComplete}
            onRemove={this.handleRemove}
            onCancel={this.toggleEdit}
          />
          : <CategoryItem category={category} onEdit={this.toggleEdit}/>
        }
      </li>
    );
  }
}
 
export default connect(
  null,
  { update, remove }
)(Category);