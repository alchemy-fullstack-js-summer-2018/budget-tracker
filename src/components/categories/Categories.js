import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Category from './Category';
import { load } from './actions';

export class Categories extends PureComponent {

  static propTypes = {
    categories: PropTypes.array,
    load: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.load();
  }

  render() { 
    const { categories } = this.props;

    return ( 
      <ul>
        {categories.map(category => (
          <Category
            key={category.key}
            category={category}
          />
        ))}
      </ul>
    );
  }
}
 
export default connect(
  state => ({ categories: state }),
  { load }
)(Categories);