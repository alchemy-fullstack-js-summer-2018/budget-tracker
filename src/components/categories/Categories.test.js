import React from 'react';
import { shallow } from 'enzyme';
import { Categories } from './Categories';
import toJSON from 'enzyme-to-json';

describe('Categories', () => {
  it('renders categories', () => {
    const categories = [
      { key: '1', name: 'a' },
      { key: '2', name: 'b' },
      { key: '3', name: 'c' }
    ];
    const load = jest.fn();
    const wrapper = shallow(
      <Categories categories={categories} load={load} />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});