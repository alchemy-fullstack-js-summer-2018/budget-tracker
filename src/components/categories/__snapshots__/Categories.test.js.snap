// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders categories 1`] = `
<ul>
  <Connect(Category)
    category={
      Object {
        "key": "1",
        "name": "a",
      }
    }
    key="1"
  />
  <Connect(Category)
    category={
      Object {
        "key": "2",
        "name": "b",
      }
    }
    key="2"
  />
  <Connect(Category)
    category={
      Object {
        "key": "3",
        "name": "c",
      }
    }
    key="3"
  />
</ul>
`;
